base/intfuncs.jl

```diff
@@ -298,7 +298,11 @@ function invmod(n::T) where {T<:BitInteger}
 end
 
 # ^ for any x supporting *
-to_power_type(x) = convert(Base._return_type(*, Tuple{typeof(x), typeof(x)}), x)
+function to_power_type(x::Number)
+    T = promote_type(typeof(x), typeof(one(x)), typeof(x*x))
```

Comment:
```suggestion
    T = promote_type(typeof(x), typeof(x*x))
```

It just occured to me that a multiplicative identity doesn't necessarily exist:

https://en.wikipedia.org/wiki/Rng_(algebra)

Are such structures supported?

Reply:
I guess it'd be OK to require a type without a multiplicative identity to implement `^` themselves?